fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DeleteResourcePersister.java

```diff
@@ -52,16 +57,68 @@ public void persist(final OCFLPersistentStorageSession session, final ResourceOp
-            // We are at the root of the object.
-            objectSession.deleteObject();
+            // We are at the root of the object, so delete all the data files.
+            try {
+                objectSession.listHeadSubpaths().filter(p -> !isSidecarSubpath(p)).forEach(
+                        p -> deletePathWrapped(p, objectSession));
+            } catch (final PersistentStorageRuntimeException exc) {
+                // Rethrow the exception as a checked exception
+                throw new PersistentStorageException(exc);
+            }
         } else {
             final var relativeSubPath = relativizeSubpath(fedoraResourceRoot, operation.getResourceId());
             final var ocflSubPath = resolveOCFLSubpath(fedoraResourceRoot, relativeSubPath);
-            final var headers = readHeaders(objectSession, ocflSubPath);
-            final var filePath = resolveExtensions(ocflSubPath,
-                    !Objects.equals(NON_RDF_SOURCE.toString(), headers.getInteractionModel())
-            );
-            objectSession.delete(filePath);
+            final var headers = (ResourceHeadersImpl) readHeaders(objectSession, ocflSubPath);
+            final boolean isRdf = !Objects.equals(NON_RDF_SOURCE.toString(), headers.getInteractionModel());
+            final var filePath = resolveExtensions(ocflSubPath, isRdf);
+            deletePath(filePath, objectSession, headers);
+            if (!isRdf) {
+                // Delete the description too.
+                final var descPath = resolveExtensions(ocflSubPath + "-description", true);
+                deletePath(descPath, objectSession);
+            }
+        }
+    }
+
+    /**
+     * Simple utility to delete a path's files and mark them as deleted in the headers file.
+     * @param path Path to delete
+     * @param session Session to delete the path in.
+     */
+    private void deletePath(final String path, final OCFLObjectSession session) throws PersistentStorageException{
+        // readHeaders and writeHeaders need the subpath where as delete needs the file name. So remove any extensions.
+        final var no_extension = (path.contains(".") ? path.substring(0, path.indexOf(".")) : path);
+        final var headers = (ResourceHeadersImpl) readHeaders(session, no_extension);
+        deletePath(path, session, headers);
+    }
+
+    /**
+     * Simple utility to delete a path's files and mark them as deleted in the headers file.
+     * @param path Path to delete
+     * @param session Session to delete the path in.
+     * @param headers The headers for the file.
+     * @throws PersistentStorageException if can't read, write or delete a file.
+     */
+    private void deletePath(final String path, final OCFLObjectSession session, final ResourceHeadersImpl headers)
+        throws PersistentStorageException {
+        session.delete(path);
+        headers.setDeleted(true);
+        headers.setLastModifiedDate(Instant.now());
```

Comment:
I think you might want to call `ResourceHeadersUtil.touchModificationHeaders()` here instead of directly setting the lastModifiedDate.